### PR TITLE
Feat/#70 search plants by difficulty and category

### PR DESCRIPTION
**Pull request by tobireith**
Basic Layout:
image

Select Categories:
image

### CODE REVIEW

File: app/src/main/res/layout/difficulty_spinner_row.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="fill_parent"
+    android:layout_height="wrap_content"
+    android:orientation="horizontal"
+    android:background="@color/primary">
+
+    <TextView
+        android:id="@+id/plant_difficulty_string"
+        android:text="Easy"
```
Comment (hoffmatteo) [inline app/src/main/res/layout/difficulty_spinner_row.xml]: Should be replaced by @string

File: app/src/main/res/layout/activity_search.xml [hunk 2]
```diff
@@ -95,10 +95,50 @@
...
+        android:text="Category"
+        app:layout_constraintStart_toStartOf="@+id/search_difficulty"
+        app:layout_constraintTop_toBottomOf="@+id/search_difficulty" />
+
+    <Button
+        android:id="@+id/test_button"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:text="SELECT CATEGORY"
```
Comment (hoffmatteo) [inline app/src/main/res/layout/activity_search.xml]: Should also be replaced by @string

File: app/src/main/res/layout/activity_search.xml [hunk 3]
```diff
@@ -95,10 +95,50 @@
...
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintVertical_bias="1.0"
         app:menu="@menu/bottom_nav_menu" />
 
+    <TextView
+        android:id="@+id/search_difficulty"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:text="Difficulty"
```
Comment (hoffmatteo) [inline app/src/main/res/layout/activity_search.xml]: Should also be replaced by @string

File: app/src/main/res/layout/activity_search.xml [hunk 4]
```diff
@@ -95,10 +95,50 @@
...
+        app:layout_constraintHorizontal_bias="1.0"
+        app:layout_constraintStart_toEndOf="@+id/search_difficulty"
+        app:layout_constraintTop_toTopOf="@+id/search_difficulty" />
 
+    <TextView
+        android:id="@+id/search_category"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="24dp"
+        android:text="Category"
```
Comment (hoffmatteo) [inline app/src/main/res/layout/activity_search.xml]: Should also be replaced by @string

**Review by hoffmatteo — COMMENTED** [review]
The categority popup menu is great!

**Review by hoffmatteo — APPROVED** [review]

File: app/src/main/java/de/othr/plantico/ui/search/SearchableActivity.kt [hunk 5]
```diff
@@ -1,92 +1,179 @@
...
-                )
-                binding.numberResultsFound.text = getString(R.string.results_found, results.size)
+                executeSearch()
                 return false
             }
         })
+        binding.bottomNavigation.setupMenuBinding(R.id.action_search, this)
 
+        //Register Listener for Dialog Button
+        binding.testButton.setOnClickListener(object : View.OnClickListener {
```
Comment (emuguy1) [inline app/src/main/java/de/othr/plantico/ui/search/SearchableActivity.kt]: maybe change the Name of the button as testButton  maybe is not the best

File: app/src/main/res/layout/activity_search.xml [hunk 6]
```diff
@@ -95,10 +95,50 @@
...
+        app:layout_constraintStart_toStartOf="@+id/search_difficulty"
+        app:layout_constraintTop_toBottomOf="@+id/search_difficulty" />
+
+    <Button
+        android:id="@+id/test_button"
```
Comment (emuguy1) [inline app/src/main/res/layout/activity_search.xml]: As said in the comment change to category button or something like that

**Review by emuguy1 — APPROVED** [review]

**Review by emuguy1 — APPROVED** [review]

**Pull request merged**